### ISSUE TITLE
Building for Windows issues

### ISSUE THREAD

**Issue by FabiFaust** [Velocidex/c-aff4#142]
Is the README.windows guide for building the code for windows systems still up to date? I have had plenty of issues trying to get it to compile by following it step by step on Debian 10.

In the last step the compilation always fails, because of **undefined reference to 'LZ4_compressBound'** , **undefined reference to 'LZ4_compress_default'** and **undefined reference to 'LZ4_decompress_safe'** .´

The exact output from running **make -j4 install-strip** was this:

```
if test -z 'i686-w64-mingw32-strip'; then \
  make  INSTALL_PROGRAM="/bin/bash /home/debian/c-aff4/config/install-sh -c -s" \
    install_sh_PROGRAM="/bin/bash /home/debian/c-aff4/config/install-sh -c -s" INSTALL_STRIP_FLAG=-s \
      install; \
else \
  make  INSTALL_PROGRAM="/bin/bash /home/debian/c-aff4/config/install-sh -c -s" \
    install_sh_PROGRAM="/bin/bash /home/debian/c-aff4/config/install-sh -c -s" INSTALL_STRIP_FLAG=-s \
    "INSTALL_PROGRAM_ENV=STRIPPROG='i686-w64-mingw32-strip'" install; \
fi
make[1]: Entering directory '/home/debian/c-aff4'
Making install in aff4
make[2]: Entering directory '/home/debian/c-aff4/aff4'
depbase=`echo aff4_image.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_image.lo -MD -MP -MF $depbase.Tpo -c -o aff4_image.lo aff4_image.cc &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo data_store.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT data_store.lo -MD -MP -MF $depbase.Tpo -c -o data_store.lo data_store.cc &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo libaff4.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT libaff4.lo -MD -MP -MF $depbase.Tpo -c -o libaff4.lo libaff4.cc &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo rdf.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT rdf.lo -MD -MP -MF $depbase.Tpo -c -o rdf.lo rdf.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_image.lo -MD -MP -MF .deps/aff4_image.Tpo -c aff4_image.cc -o aff4_image.o
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT rdf.lo -MD -MP -MF .deps/rdf.Tpo -c rdf.cc -o rdf.o
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT data_store.lo -MD -MP -MF .deps/data_store.Tpo -c data_store.cc -o data_store.o
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT libaff4.lo -MD -MP -MF .deps/libaff4.Tpo -c libaff4.cc -o libaff4.o
depbase=`echo zip.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT zip.lo -MD -MP -MF $depbase.Tpo -c -o zip.lo zip.cc &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo aff4_imager_utils.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_imager_utils.lo -MD -MP -MF $depbase.Tpo -c -o aff4_imager_utils.lo aff4_imager_utils.cc &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo aff4_map.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_map.lo -MD -MP -MF $depbase.Tpo -c -o aff4_map.lo aff4_map.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_imager_utils.lo -MD -MP -MF .deps/aff4_imager_utils.Tpo -c aff4_imager_utils.cc -o aff4_imager_utils.o
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT zip.lo -MD -MP -MF .deps/zip.Tpo -c zip.cc -o zip.o
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_map.lo -MD -MP -MF .deps/aff4_map.Tpo -c aff4_map.cc -o aff4_map.o
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4_imager_utils.cc:7:
/home/scudette/build/mingw//include/tclap/sstream.h:31: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:30,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4_imager_utils.cc:7:
./config.h:56: note: this is the location of the previous definition
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/StandardTraits.h:30,
                 from /home/scudette/build/mingw//include/tclap/Arg.h:56,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4_imager_utils.cc:7:
./config.h:56: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4_imager_utils.cc:7:
/home/scudette/build/mingw//include/tclap/sstream.h:31: note: this is the location of the previous definition
 #define HAVE_SSTREAM
 
depbase=`echo lexicon.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT lexicon.lo -MD -MP -MF $depbase.Tpo -c -o lexicon.lo lexicon.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT lexicon.lo -MD -MP -MF .deps/lexicon.Tpo -c lexicon.cc -o lexicon.o
depbase=`echo aff4_directory.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_directory.lo -MD -MP -MF $depbase.Tpo -c -o aff4_directory.lo aff4_directory.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_directory.lo -MD -MP -MF .deps/aff4_directory.Tpo -c aff4_directory.cc -o aff4_directory.o
depbase=`echo aff4_file.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_file.lo -MD -MP -MF $depbase.Tpo -c -o aff4_file.lo aff4_file.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_file.lo -MD -MP -MF .deps/aff4_file.Tpo -c aff4_file.cc -o aff4_file.o
depbase=`echo aff4_symstream.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4_symstream.lo -MD -MP -MF $depbase.Tpo -c -o aff4_symstream.lo aff4_symstream.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT aff4_symstream.lo -MD -MP -MF .deps/aff4_symstream.Tpo -c aff4_symstream.cc -o aff4_symstream.o
depbase=`echo libaff4-c.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT libaff4-c.lo -MD -MP -MF $depbase.Tpo -c -o libaff4-c.lo libaff4-c.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT libaff4-c.lo -MD -MP -MF .deps/libaff4-c.Tpo -c libaff4-c.cc -o libaff4-c.o
depbase=`echo volume_group.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT volume_group.lo -MD -MP -MF $depbase.Tpo -c -o volume_group.lo volume_group.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT volume_group.lo -MD -MP -MF .deps/volume_group.Tpo -c volume_group.cc -o volume_group.o
depbase=`echo tclap_parsers.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool  --tag=CXX   --mode=compile i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT tclap_parsers.lo -MD -MP -MF $depbase.Tpo -c -o tclap_parsers.lo tclap_parsers.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I. -I.. -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -MT tclap_parsers.lo -MD -MP -MF .deps/tclap_parsers.Tpo -c tclap_parsers.cc -o tclap_parsers.o
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from tclap_parsers.cc:1:
/home/scudette/build/mingw//include/tclap/sstream.h:31: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:30,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from tclap_parsers.cc:1:
./config.h:56: note: this is the location of the previous definition
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/StandardTraits.h:30,
                 from /home/scudette/build/mingw//include/tclap/Arg.h:56,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from tclap_parsers.cc:1:
./config.h:56: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from tclap_parsers.cc:1:
/home/scudette/build/mingw//include/tclap/sstream.h:31: note: this is the location of the previous definition
 #define HAVE_SSTREAM
 
depbase=`echo aff4imager.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
i686-w64-mingw32-g++ -DHAVE_CONFIG_H -I.  -I..  -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -MT aff4imager.o -MD -MP -MF $depbase.Tpo -c -o aff4imager.o aff4imager.cc &&\
mv -f $depbase.Tpo $depbase.Po
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4imager.cc:21:
/home/scudette/build/mingw//include/tclap/sstream.h:31: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:30,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4imager.cc:21:
./config.h:56: note: this is the location of the previous definition
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/StandardTraits.h:30,
                 from /home/scudette/build/mingw//include/tclap/Arg.h:56,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4imager.cc:21:
./config.h:56: warning: "HAVE_SSTREAM" redefined
 #define HAVE_SSTREAM 1
 
In file included from /home/scudette/build/mingw//include/tclap/Arg.h:40,
                 from /home/scudette/build/mingw//include/tclap/SwitchArg.h:30,
                 from /home/scudette/build/mingw//include/tclap/CmdLine.h:27,
                 from ../aff4/tclap_parsers.h:8,
                 from ../aff4/aff4_imager_utils.h:30,
                 from aff4imager.cc:21:
/home/scudette/build/mingw//include/tclap/sstream.h:31: note: this is the location of the previous definition
 #define HAVE_SSTREAM
 
/bin/bash ../libtool  --tag=CXX   --mode=link i686-w64-mingw32-g++ -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -static -static-libstdc++ -static-libgcc -Wl,-Bstatic,--whole-archive,-lpthread -Wl,--no-whole-archive -L/home/scudette/build/mingw//lib -static -static-libstdc++ -o libaff4.la -rpath /home/scudette/build/mingw/lib aff4_image.lo aff4_init.lo data_store.lo libaff4.lo rdf.lo zip.lo aff4_imager_utils.lo aff4_map.lo lexicon.lo aff4_directory.lo aff4_file.lo aff4_symstream.lo libaff4-c.lo volume_group.lo tclap_parsers.lo -lrpcrt4 -lshlwapi -luriparser -lpthread -llz4 -lsnappy   -L/home/scudette/build/mingw//lib -lz -L/home/scudette/build/mingw/lib -lraptor2  -luriparser -lpthread -llz4 -lsnappy 
libtool: warning: undefined symbols not allowed in i686-w64-mingw32 shared libraries; building static only
libtool: link: rm -fr  .libs/libaff4.a .libs/libaff4.la .libs/libaff4.lai
libtool: link: i686-w64-mingw32-ar cru .libs/libaff4.a  aff4_image.o aff4_init.o data_store.o libaff4.o rdf.o zip.o aff4_imager_utils.o aff4_map.o lexicon.o aff4_directory.o aff4_file.o aff4_symstream.o libaff4-c.o volume_group.o tclap_parsers.o
i686-w64-mingw32-ar: `u' modifier ignored since `D' is the default (see `U')
libtool: link: i686-w64-mingw32-ranlib .libs/libaff4.a
libtool: link: ( cd ".libs" && rm -f "libaff4.la" && ln -s "../libaff4.la" "libaff4.la" )
/bin/bash ../libtool  --tag=CXX   --mode=link i686-w64-mingw32-g++ -mno-ms-bitfields -static -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare  -s -static -static-libstdc++ -static-libgcc -Wl,-Bstatic,--whole-archive,-lpthread -Wl,--no-whole-archive -L/home/scudette/build/mingw//lib -static -static-libstdc++ -o aff4imager.exe aff4imager.o libaff4.la -luriparser -lpthread -llz4 -lsnappy   -L/home/scudette/build/mingw//lib -lz -L/home/scudette/build/mingw/lib -lraptor2   -luriparser -lpthread -llz4 -lsnappy 
libtool: link: i686-w64-mingw32-g++ -mno-ms-bitfields -mnop-fun-dllimport -I/home/scudette/build/mingw//include/ -DRAPTOR_STATIC -Wall -Wextra -Wpedantic -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -s -static-libstdc++ -static-libgcc -Wl,-Bstatic -Wl,--whole-archive -Wl,-lpthread -Wl,--no-whole-archive -static-libstdc++ -o aff4imager.exe aff4imager.o  -L/home/scudette/build/mingw//lib ./.libs/libaff4.a -lrpcrt4 -lshlwapi -L/home/scudette/build/mingw/lib -L/lib -lz /home/scudette/build/mingw/lib/libraptor2.a /lib/liburiparser.a -lpthread -llz4 /home/scudette/build/mingw/lib/libsnappy.a
/usr/bin/i686-w64-mingw32-ld: ./.libs/libaff4.a(aff4_image.o):aff4_image.cc:(.text+0xbae): undefined reference to `LZ4_compressBound'
/usr/bin/i686-w64-mingw32-ld: ./.libs/libaff4.a(aff4_image.o):aff4_image.cc:(.text+0xc08): undefined reference to `LZ4_compress_default'
/usr/bin/i686-w64-mingw32-ld: ./.libs/libaff4.a(aff4_image.o):aff4_image.cc:(.text+0xc89): undefined reference to `LZ4_decompress_safe'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:613: aff4imager.exe] Error 1
make[2]: Leaving directory '/home/debian/c-aff4/aff4'
make[1]: *** [Makefile:465: install-recursive] Error 1
make[1]: Leaving directory '/home/debian/c-aff4'
make: *** [Makefile:773: install-strip] Error 2

```

Basically all libraries and headers should be in the /home/scudette/build/mingw folder, however lz4 is by default installed into usr/local and was not found by ./configure  , only after dropping the library and header files into the mingw directories was it successful.  The compilation always fails with the error message above though and I've been unable to work out the reason. 

I've attached the scudette directory containing all the dependencies, maybe something is missing?
[scudette.zip](https://github.com/Velocidex/c-aff4/files/4552288/scudette.zip)

Any help in getting this to compile would be very appreciated.
